src/Microsoft.Net.Http.Headers/MediaTypeHeaderValue.cs

```diff
@@ -170,55 +205,141 @@ public IList<NameValueHeaderValue> Parameters
             }
         }
 
+        /// <summary>
+        /// Gets the media type of <see cref="MediaTypeHeaderValue"/>
+        /// </summary>
+        /// <example>
+        /// For the media type <c>"application/json"</c>, the property gives the value
+        /// <c>"application/json"</c>.
+        /// </example>
         public StringSegment MediaType
         {
             get { return _mediaType; }
             set
             {
                 HeaderUtilities.ThrowIfReadOnly(IsReadOnly);
-                CheckMediaTypeFormat(value, "value");
+                CheckMediaTypeFormat(value, ValueString);
                 _mediaType = value;
             }
         }
 
+        /// <summary>
+        /// Gets the type of the <see cref="MediaTypeHeaderValue"/>.
+        /// </summary>
+        /// <example>
+        /// For the media type <c>"application/json"</c>, the property gives the value <c>"application"</c>.
+        /// </example>
         public StringSegment Type
         {
             get
             {
-                return _mediaType.Subsegment(0, _mediaType.IndexOf('/'));
+                return _mediaType.Subsegment(0, _mediaType.IndexOf(ForwardSlashCharacter));
             }
         }
 
+        /// <summary>
+        /// Gets the subtype of the <see cref="MediaTypeHeaderValue"/>.
+        /// </summary>
+        /// <example>
+        /// For the media type <c>"application/vnd.example+json"</c>, the property gives the value
+        /// <c>"vnd.example+json"</c>.
+        /// </example>
         public StringSegment SubType
         {
             get
             {
-                return _mediaType.Subsegment(_mediaType.IndexOf('/') + 1);
+                return _mediaType.Subsegment(_mediaType.IndexOf(ForwardSlashCharacter) + 1);
+            }
+        }
+
+        /// <summary>
+        /// Gets the subtype of the <see cref="MediaTypeHeaderValue"/>, excluding any structured syntax suffix.
+        /// </summary>
+        /// <example>
+        /// For the media type <c>"application/vnd.example+json"</c>, the property gives the value
+        /// <c>"vnd.example"</c>.
+        /// </example>
+        public StringSegment SubTypeWithoutSuffix
```

Comment:
We would like to keep the functionality of Subtype the same (currently includes the suffix too). Mvc did the same thing with [MediaType](https://github.com/aspnet/Mvc/blob/dev/src/Microsoft.AspNetCore.Mvc.Core/Formatters/MediaType.cs#L211-L229)